### PR TITLE
remove unused fields

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@myidpt ptal

**Comment by hzxuzhonghu** [comment]
kindly ping @istio/wg-security-maintainers

**Comment by hzxuzhonghu** [comment]
@myidpt @incfly ptal

File: pkg/istio-agent/sds-agent.go [hunk 1]
```diff
@@ -173,26 +162,20 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig,
...
 
 	if sa.secOpts.CAEndpoint == "" {
 		// if not set, we will fallback to the discovery address
 		sa.secOpts.CAEndpoint = discAddr
 	}
-
 	// Next to the envoy config, writeable dir (mounted as mem)
-	sa.secOpts.WorkloadUDSPath = LocalSDS
-	sa.secOpts.CertsDir = sa.CertsPath
+	if sa.secOpts.WorkloadUDSPath == "" {
```
Comment (incfly) [inline pkg/istio-agent/sds-agent.go:176]: Is this introducing a different behavior?

File: security/pkg/nodeagent/secretfetcher/secretfetcher.go [hunk 2]
```diff
@@ -77,9 +77,8 @@ var (
 
 // SecretFetcher fetches secret via watching k8s secrets or sending CSR to CA.
 type SecretFetcher struct {
-	// If UseCaClient is true, use caClient to send CSR to CA.
-	UseCaClient bool
-	CaClient    security.Client
+	// If caClient is set, use caClient to send CSR to CA.
```
Comment (incfly) [inline security/pkg/nodeagent/secretfetcher/secretfetcher.go:80]: s/caClient/CaClient

**Review by incfly — COMMENTED** [review]
mostly LGTM.

File: pkg/istio-agent/sds-agent.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/istio-agent/sds-agent.go:176]: Actually, i think it is a bug ignoring `sa.secOpts.WorkloadUDSPath`

**Pull request merged**